src/typechecker/FStar.TypeChecker.Core.fst

```diff
@@ -1018,6 +1025,7 @@ let rec check_relation (g:env) (rel:relation) (t0 t1:typ)
       | Tm_abs([b0], body0, _), Tm_abs([b1], body1, _) ->
         check_relation g EQUALITY b0.binder_bv.sort b1.binder_bv.sort;!
```

Comment:
Not related to this PR, but surprised that for lambdas, we are checking binder bv sorts equality, irrespective of the input relation; whereas for arrows we check relation (with contravariance).